apps/bluetooth/app.js

```diff
@@ -224,31 +291,49 @@ module.exports = function (activity) {
 
   function onConnectionStateChangedListener (mode, state, device) {
     logger.debug(`${mode} onConnectionStateChanged(${state})`)
-
+    cancelTimer()
     if (mode === protocol.A2DP_MODE.SOURCE) {
-      var data = device != null ? {'currentDevice': device} : null
+      var data = ((device != null) ? {'currentDevice': device} : null)
       sendMsgToApp(state, data)
     }
+    if (mode !== a2dp.getMode()) {
+      logger.warn('Suppress old mode event to avoid confusing users.')
+      return
+    }
     switch (state) {
       case protocol.CONNECTION_STATE.CONNECTED:
-        speak(getText(mode, 'CONNECED_BEGIN') + device.name, res.AUDIO[state])
+        if (lastIntent === 'bluetooth_start_bluetooth_music') {
+          setTimer(() => {
+            if (a2dp.isConnected() && !a2dp.isPlaying()) {
+              var dev = a2dp.getConnectedDevice()
+              if (dev != null) {
+                speak(getText('PLAY_FAILED_ARG1S', dev.name))
+              }
+            }
+          }, 9000)
```

Comment:
Make it configurable

Reply:
use config.json to define configurable constants.